### PR TITLE
Give user option to upload files as binary in USS tree

### PR DESCRIPTION
**Pull request by likhithanimma1**
## Proposed changes

Provide users with the option to upload binary files by implementing a "Upload Files (Binary)" right-click option in the USS tree.

## Release Notes

Milestone:

Changelog:

## Types of changes

What types of changes does your code introduce to Zowe Explorer?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Updates to Documentation or Tests (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This checklist will be used as reference for both the contributor and the reviewer_

- [ ] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/zowe-explorer-vscode/wiki/Best-Practices:-Contributor-Guidance) wiki
- [ ] PR title follows [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0-beta.2/)
- [x] PR Description is included
- [ ] gif or screenshot is included if visual changes are made
- [x] `yarn workspace vscode-extension-for-zowe vscode:prepublish` has been executed
- [ ] All checks have passed (DCO, Jenkins and Code Coverage)
- [x] I have added unit test and it is passing
- [ ] I have added integration test and it is passing
- [ ] There is coverage for the code that I have added
- [ ] I have tested it manually and there are no regressions found
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any PR dependencies have been merged and published (if appropriate)

### CODE REVIEW

**Comment by traeok** [comment]
I was looking around and found #1956 - I think this PR will implement the request from that issue 👍

**Review by JillieBeanSim — CHANGES_REQUESTED** [review]
Thanks @likhithanimma1 this is looking good.  I did leave a comment on small change with naming convention of variables and I also noticed that the tree isn't updated once the upload is complete.  There is an open [issue](https://github.com/zowe/zowe-explorer-vscode/issues/2841) for the lack of upload progress being shown, I would have to double check with released to see if the tree update after upload is pre-existing.

Update: the tree not updating after upload is a pre-existing issue seen in marketplace, created issue https://github.com/zowe/zowe-explorer-vscode/issues/2849

File: packages/zowe-explorer/src/uss/actions.ts [hunk 1]
```diff
@@ -146,9 +146,8 @@ export async function uploadDialog(node: IZoweUSSTreeNode, ussFileProvider: IZow
 
     await Promise.all(
         value.map(async (item) => {
-            const isBinary = isBinaryFileSync(item.fsPath);
-
-            if (isBinary) {
+            const Binary = isBinaryFileSync(item.fsPath);
```
[the remark this answers is not in the document]
Comment (likhithanimma1) [reply packages/zowe-explorer/src/uss/actions.ts]: Changed it and added indication for action happening using withProgress #2841  and also tree automatically updating after upload complete #2849

**Review by traeok — CHANGES_REQUESTED** [review]
Thanks for working on this enhancement @likhithanimma1!

I noticed that two status bar messages appear at the same time as the upload:
- "Uploading file to USS tree"
- "Upload action was cancelled"

After the "Uploading file..." message disappears, it seems the file was not successfully uploaded as I cannot fetch the file in the destination folder afterward:

ZE-USS-BinaryUpload

**Comment by likhithanimma1** [comment]
> Thanks for working on this enhancement @likhithanimma1!
> 
> I noticed that two status bar messages appear at the same time as the upload:
> 
> * "Uploading file to USS tree"
> * "Upload action was cancelled"
> 
> After the "Uploading file..." message disappears, it seems the file was not successfully uploaded as I cannot fetch the file in the destination folder afterward:
> 
> ZE-USS-BinaryUpload [ ZE-USS-BinaryUpload ](https://private-user-images.githubusercontent.com/86500639/322951467-49ec0c35-c658-464d-8cad-4fe353786f56.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTMzMzk1MjAsIm5iZiI6MTcxMzMzOTIyMCwicGF0aCI6Ii84NjUwMDYzOS8zMjI5NTE0NjctNDllYzBjMzUtYzY1OC00NjRkLThjYWQtNGZlMzUzNzg2ZjU2LmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDE3VDA3MzM0MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQwYzYxYjYzYzVhY2NjNmMwZDVhMDcxNWVmNDI1MGNjN2Y2MzNmYzI3MDI2MmI2NDVmNjJlYTAyNTY2YTE1NGYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.ku3h2iYuwdeAu8LIq82dLp8shruw8rA0dxbB2NRyNtQ) [ ](https://private-user-images.githubusercontent.com/86500639/322951467-49ec0c35-c658-464d-8cad-4fe353786f56.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTMzMzk1MjAsIm5iZiI6MTcxMzMzOTIyMCwicGF0aCI6Ii84NjUwMDYzOS8zMjI5NTE0NjctNDllYzBjMzUtYzY1OC00NjRkLThjYWQtNGZlMzUzNzg2ZjU2LmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDE3VDA3MzM0MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQwYzYxYjYzYzVhY2NjNmMwZDVhMDcxNWVmNDI1MGNjN2Y2MzNmYzI3MDI2MmI2NDVmNjJlYTAyNTY2YTE1NGYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.ku3h2iYuwdeAu8LIq82dLp8shruw8rA0dxbB2NRyNtQ)

Addressed the behaviour!

**Review by traeok — COMMENTED** [review]
I can now upload files as binary and the progress bar UX looks good, the tree views show the binary indicator as expected - thanks Likhitha for addressing that!

I tried deleting an uploaded file through Zowe Explorer afterward, but I saw that the delete dialog showed an empty name. When I click Delete in the dialog, the dialog disappears and the file is not deleted. The only workarounds I've found so far is to either reload the window or change the search path so that it has to re-create the node.

ZE-USS-DeleteAfterBinaryUpload

I'm not sure if this is specific to this feature, or if it also occurs with the regular "Upload Files" operation. I'll run some tests to confirm - if it's happening with the standard upload we can file this behavior as a separate issue.

**Comment by JillieBeanSim** [comment]
Thanks for addressing the withProgress and tree refresh, this is looking good.  I did experience the same issue seen as Trae with the filename missing from delete dialog but only with the upload of binary, same scenario works fine with regular upload.

**Comment by likhithanimma1** [comment]
> Thanks for addressing the withProgress and tree refresh, this is looking good. I did experience the same issue seen as Trae with the filename missing from delete dialog but only with the upload of binary, same scenario works fine with regular upload.

Addressed deletion of files after uploading @JillieBeanSim

**Review by traeok — APPROVED** [review]
LGTM, thanks @likhithanimma1!

**Review by JillieBeanSim — APPROVED** [review]
LGTM! thanks @likhithanimma1 for this enhancement ❤️

**Pull request merged**